### PR TITLE
Fixing build errors in Windows for lib/ATen

### PR DESCRIPTION
**Pull request by peterjc123**
Fixing build errors in Windows for lib/ATen and MSVC

### CODE REVIEW

File: torch/lib/ATen/Formatting.h [hunk 1]
```diff
@@ -6,11 +6,10 @@
 
 namespace at {
 
-
-std::ostream& operator<<(std::ostream & out, IntList list);
-std::ostream& operator<<(std::ostream & out, Backend b);
-std::ostream& operator<<(std::ostream & out, ScalarType t);
-std::ostream& print(std::ostream& stream, const Tensor & tensor, int64_t linesize);
+ATen_CLASS std::ostream& operator<<(std::ostream & out, IntList list);
```
Comment (zdevito) [inline torch/lib/ATen/Formatting.h]: Can we rename this ATen_EXPORT if we are going to use it to export functions?

File: torch/lib/ATen/templates/Type.h [hunk 2]
```diff
@@ -3,10 +3,17 @@
 #include <memory>
 #include <limits>
 
+#include "ATen/ATenGeneral.h"
 #include "ATen/ArrayRef.h"
 #include "ATen/Half.h"
 #include "ATen/SparseTensorRef.h"
 
+#ifdef _MSC_VER
```
Comment (zdevito) [inline torch/lib/ATen/templates/Type.h]: Can we add a comment why this thing is necessary?

File: torch/lib/ATen/test/test_assert.h [hunk 3]
```diff
@@ -11,14 +11,20 @@ static inline void barf(const char *fmt, ...) {
   throw std::runtime_error(msg);
 }
 
+#if defined(__GNUC__) || defined(__ICL) || defined(__clang__)
+#define EXPECT(x, y) (__builtin_expect((x),(y)))
```
Comment (zdevito) [inline torch/lib/ATen/test/test_assert.h]: EXPECT -> AT_EXPECT, to avoid namespacing problems.

**Review by zdevito — COMMENTED** [review]
just minor naming nits.

File: torch/lib/ATen/Formatting.h (hunk 1 above)
Comment (peterjc123) [reply torch/lib/ATen/Formatting.h]: ATen_EXPORT is defined while building using cmake, so we have to use another name.

Comment (peterjc123) [reply torch/lib/ATen/Formatting.h]: For export functions, there's ATen_API. It can be used for export C-style functions.

**Comment by soumith** [comment]
I dont see `ATen_CLASS` being defined anywhere. Where is it defined?

**Comment by peterjc123** [comment]
lt's defined in ATen_General.h.

**Comment by soumith** [comment]
this is now merged into master.

**Pull request closed without merging**